local Corrections = {
    -- Basic corrections
    ["i"] = "I",
    ["im"] = "I'm",
    ["ive"] = "I've",
    ["id"] = "I'd",
    ["ill"] = "I'll",
    
    -- Gaming/Internet Slang
    ["ez"] = "easy",
    ["ezy"] = "easy",
    ["ezpz"] = "easy peasy",
    ["n00b"] = "noob",
    ["nub"] = "noob",
    ["newb"] = "noob",
    ["pwn"] = "own",
    ["pwned"] = "owned",
    ["rekt"] = "wrecked",
    ["rekd"] = "wrecked",
    ["rek"] = "wreck",
    ["git"] = "get",
    ["gits"] = "gets",
    ["gitin"] = "getting",
    ["gittin"] = "getting",
    ["hax"] = "hacks",
    ["haxor"] = "hacker",
    ["hackz"] = "hacks",
    ["h4x"] = "hacks",
    ["pro"] = "professional",
    ["proz"] = "professionals",
    ["skrub"] = "scrub",
    ["skrublord"] = "scrublord",
    ["leet"] = "elite",
    ["l33t"] = "elite",
    ["1337"] = "elite",
    
    -- Numbers as Letters
    ["m3"] = "me",
    ["u2"] = "you too",
    ["r3kt"] = "wrecked",
    ["n1"] = "nice one",
    ["2u2"] = "to you too",
    ["4u"] = "for you",
    ["m8t"] = "mate",
    
    -- Emotional/Reaction
    ["lel"] = "lol",
    ["kek"] = "lol",
    ["lawl"] = "lol",
    ["lulz"] = "lols",
    ["rofl"] = "rolling on floor laughing",
    ["rofls"] = "rolling on floor laughing",
    ["roflmao"] = "rolling on floor laughing my ass off",
    ["lmfao"] = "laughing my fucking ass off",
    ["haha"] = "ha ha",
    ["hehe"] = "he he",
    ["xd"] = "XD",
    ["dx"] = "DX",
    ["omfg"] = "oh my fucking god",
    ["wtf"] = "what the fuck",
    ["tf"] = "the fuck",
    ["ffs"] = "for fuck's sake",
    ["smh"] = "shaking my head",
    ["nvm"] = "never mind",
    ["nvmd"] = "never mind",
    ["nm"] = "never mind",
    ["tbf"] = "to be fair",
    ["tbh"] = "to be honest",
    
    -- Additional Common Misspellings
    ["recieve"] = "receive",
    ["wierd"] = "weird",
    ["peice"] = "piece",
    ["freind"] = "friend",
    ["beleive"] = "believe",
    ["occured"] = "occurred",
    ["tommorow"] = "tomorrow",
    ["tommorrow"] = "tomorrow",
    ["tomorro"] = "tomorrow",
    ["seperate"] = "separate",
    ["definately"] = "definitely",
    ["alot"] = "a lot",
    ["allready"] = "already",
    ["untill"] = "until",
    ["comming"] = "coming",
    ["begining"] = "beginning",
    ["truely"] = "truly",
    ["truley"] = "truly",
    ["thier"] = "their",
    ["theyre"] = "they're",
    ["loose"] = "lose",
    ["loosing"] = "losing",
    ["whos"] = "who's",
    ["whose"] = "who's",
    ["thats"] = "that's",
    ["wheres"] = "where's",
    ["heres"] = "here's",
    ["youre"] = "you're",
    ["there"] = "they're",
    ["wont"] = "won't",
    ["cant"] = "can't",
    ["dont"] = "don't",
    ["didnt"] = "didn't",
    ["wasnt"] = "wasn't",
    ["werent"] = "weren't",
    ["hasnt"] = "hasn't",
    ["havent"] = "haven't",
    ["wouldnt"] = "wouldn't",
    ["shouldnt"] = "shouldn't",
    ["couldnt"] = "couldn't",
    ["isnt"] = "isn't",
    ["arent"] = "aren't",
    ["aint"] = "ain't",
        ["teh"] = "the",
    ["hte"] = "the",
    ["thw"] = "the",
    ["tge"] = "the",
    ["rhe"] = "the",
    ["tue"] = "the",
    ["yhe"] = "the",
    ["rhe"] = "the",
    ["thr"] = "the",
    ["ther"] = "there",
    ["thre"] = "there",
    ["thier"] = "their",
    ["thire"] = "their",
    ["thisr"] = "their",
    ["theys"] = "they",
    ["yhey"] = "they",
    ["tjey"] = "they",
    ["thry"] = "they",
    ["taht"] = "that",
    ["thet"] = "that",
    ["thar"] = "that",
    ["thst"] = "that",
    ["thay"] = "that",
    ["thst"] = "that",
    ["wiht"] = "with",
    ["whit"] = "with",
    ["wirh"] = "with",
    ["woth"] = "with",
    ["whth"] = "with",
    ["wuth"] = "with",
    ["adn"] = "and",
    ["nad"] = "and",
    ["anf"] = "and",
    ["abd"] = "and",
    ["ans"] = "and",
    ["snd"] = "and",
    ["andd"] = "and",
    ["ahd"] = "and",
    ["yuo"] = "you",
    ["yoy"] = "you",
    ["ypu"] = "you",
    ["yiu"] = "you",
    ["uou"] = "you",
    ["yku"] = "you",
    ["uoy"] = "you",
    ["youu"] = "you",
    ["yoou"] = "you",
    ["havw"] = "have",
    ["hsve"] = "have",
    ["havr"] = "have",
    ["habe"] = "have",
    ["hqve"] = "have",
    ["hage"] = "have",
    ["haev"] = "have",
    ["hvae"] = "have",
    ["waht"] = "what",
    ["whta"] = "what",
    ["whar"] = "what",
    ["whst"] = "what",
    ["wgat"] = "what",
    ["wuat"] = "what",
    ["whay"] = "what",
    ["whatt"] = "what",
    ["whhat"] = "what",
    ["wnat"] = "want",
    ["wamt"] = "want",
    ["wany"] = "want",
    ["wanr"] = "want",
    ["wamt"] = "want",
    ["wsnt"] = "want",
    ["wannt"] = "want",
    ["wanntt"] = "want",
    ["thsi"] = "this",
    ["tihs"] = "this",
    ["tjis"] = "this",
    ["yhis"] = "this",
    ["thus"] = "this",
    ["thid"] = "this",
    ["rhis"] = "this",
    ["thiis"] = "this",
    ["thiss"] = "this",
    ["jsut"] = "just",
    ["jstu"] = "just",
    ["kust"] = "just",
    ["judt"] = "just",
    ["juat"] = "just",
    ["jist"] = "just",
    ["jusy"] = "just",
    ["jusst"] = "just",
    ["justt"] = "just",
    ["liek"] = "like",
    ["liek"] = "like",
    ["likw"] = "like",
    ["likr"] = "like",
    ["linke"] = "like",
    ["kike"] = "like",
    ["lige"] = "like",
    ["likee"] = "like",
    ["likke"] = "like",
    ["knwo"] = "know",
    ["konw"] = "know",
    ["kmow"] = "know",
    ["knoww"] = "know",
    ["knoe"] = "know",
    ["kbow"] = "know",
    ["knoww"] = "know",
    ["knowww"] = "know",
    ["thnik"] = "think",
    ["thiml"] = "think",
    ["thimk"] = "think",
    ["yhink"] = "think",
    ["thibk"] = "think",
    ["thunk"] = "think",
    ["thinkk"] = "think",
    ["thinck"] = "think",
    ["becuase"] = "because",
    ["becasue"] = "because",
    ["beacuse"] = "because",
    ["becuze"] = "because",
    ["becouse"] = "because",
    ["becayse"] = "because",
    ["becaude"] = "because",
    ["becausee"] = "because",
    ["becausese"] = "because",
    ["abuot"] = "about",
    ["abotu"] = "about",
    ["abouy"] = "about",
    ["abput"] = "about",
    ["abiut"] = "about",
    ["aboutt"] = "about",
    ["abbout"] = "about",
    ["aboutt"] = "about",
    ["peopel"] = "people",
    ["poeple"] = "people",
    ["peiple"] = "people",
    ["peolle"] = "people",
    ["pwople"] = "people",
    ["peopple"] = "people",
    ["peoplee"] = "people",
    ["peeopple"] = "people",

    -- Double Letter Mistakes
    ["realy"] = "really",
    ["relly"] = "really",
    ["realyy"] = "really",
    ["reallly"] = "really",
    ["realley"] = "really",
    ["rreally"] = "really",
    ["reallyy"] = "really",
    ["finaly"] = "finally",
    ["finelly"] = "finally",
    ["finallly"] = "finally",
    ["finalley"] = "finally",
    ["finnally"] = "finally",
    ["finalyy"] = "finally",
    ["basicaly"] = "basically",
    ["basicly"] = "basically",
    ["basicalli"] = "basically",
    ["basickly"] = "basically",
    ["basiccally"] = "basically",
    ["basicallyy"] = "basically",
            ["comming"] = "coming",
        ["runing"] = "running",
        ["geting"] = "getting",
        ["going"] = "going",
        ["speking"] = "speaking",
        ["thinkin"] = "thinking",
        ["talkin"] = "talking",
        ["writin"] = "writing",
        ["playin"] = "playing",
        ["doin"] = "doing",
                ["dissapoint"] = "disappoint",
        ["dissapear"] = "disappear",
        ["dissagree"] = "disagree",
        ["dissable"] = "disable",
        ["dissconnect"] = "disconnect",
    -- Common chat abbreviations
    ["ik"] = "I know",
    ["idk"] = "I don't know",
    ["idek"] = "I don't even know",
    ["idrk"] = "I don't really know",
    ["iirc"] = "if I remember correctly",
    ["tbh"] = "to be honest",
    ["ngl"] = "not gonna lie",
    ["omg"] = "oh my god",
    ["omw"] = "on my way",
    ["brb"] = "be right back",
    ["gtg"] = "got to go",
    ["afk"] = "away from keyboard",
    ["nvm"] = "never mind",
    ["wym"] = "what you mean",
    ["wdym"] = "what do you mean",
    
    -- Thank you variations
    ["ty"] = "thank you",
    ["tyvm"] = "thank you very much",
    ["tysm"] = "thank you so much",
    ["thx"] = "thanks",
        -- Common Keyboard Typos
    ["teh"] = "the",
    ["adn"] = "and",
    ["nad"] = "and",
    ["anf"] = "and",
    ["ans"] = "and",
    ["tje"] = "the",
    ["yhe"] = "the",
    ["hte"] = "the",
    ["thr"] = "the",
    ["jsut"] = "just",
    ["jugst"] = "just",
    ["jsut"] = "just",
    ["justr"] = "just",
    ["waht"] = "what",
    ["whta"] = "what",
    ["wahts"] = "what's",
    ["whast"] = "what's",
    ["thign"] = "thing",
    ["thigg"] = "thing",
    ["thigns"] = "things",
    ["liek"] = "like",
    ["liek"] = "like",
    ["leik"] = "like",
    ["lke"] = "like",
    ["liekd"] = "liked",
    ["becuase"] = "because",
    ["becasue"] = "because",
    ["becuz"] = "because",
    ["becoz"] = "because",
    ["bcos"] = "because",
    ["becouse"] = "because",
    ["becz"] = "because",
    ["beacuse"] = "because",
    ["becuse"] = "because",
    ["becase"] = "because",
    
    -- More Gaming/Internet Slang
    ["poggers"] = "amazing",
    ["pog"] = "amazing",
    ["pogchamp"] = "champion",
    ["pepega"] = "stupid",
    ["kappa"] = "joking",
    ["sadge"] = "sad",
    ["monkas"] = "nervous",
    ["sheesh"] = "wow",
    ["sheeesh"] = "wow",
    ["sheeeesh"] = "wow",
    ["bussin"] = "amazing",
    ["bussin'"] = "amazing",
    ["bussin"] = "amazing",
    ["cap"] = "lie",
    ["nocap"] = "no lie",
    ["no cap"] = "no lie",
    ["fr fr"] = "for real",
    ["frfr"] = "for real",
    ["ong"] = "on god",
    ["fax"] = "facts",
    ["faxx"] = "facts",
    ["faxxx"] = "facts",
    
    -- More Common Typos
    ["doesnt"] = "doesn't",
    ["dont"] = "don't",
    ["cant"] = "can't",
    ["wont"] = "won't",
    ["shouldnt"] = "shouldn't",
    ["couldnt"] = "couldn't",
    ["wouldnt"] = "wouldn't",
    ["didnt"] = "didn't",
    ["hadnt"] = "hadn't",
    ["hasnt"] = "hasn't",
    ["havent"] = "haven't",
    ["wasnt"] = "wasn't",
    ["werent"] = "weren't",
    ["arent"] = "aren't",
    ["isnt"] = "isn't",
    ["aint"] = "ain't",
    ["im"] = "I'm",
    ["ill"] = "I'll",
    ["id"] = "I'd",
    ["ive"] = "I've",
    ["youve"] = "you've",
    ["youll"] = "you'll",
    ["youre"] = "you're",
    ["theyre"] = "they're",
    ["theyve"] = "they've",
    ["theyll"] = "they'll",
    ["weve"] = "we've",
    ["well"] = "we'll",
    ["were"] = "we're",
    
    -- More Informal Speech
    ["yall"] = "you all",
    ["y'all"] = "you all",
    ["ya'll"] = "you all",
    ["yal"] = "you all",
    ["yuh"] = "yes",
    ["yeh"] = "yes",
    ["yee"] = "yes",
    ["ye"] = "yes",
    ["ya"] = "yes",
    ["nah"] = "no",
    ["naw"] = "no",
    ["nop"] = "nope",
    ["nahh"] = "no",
    ["naww"] = "no",
    ["yup"] = "yes",
    ["yupp"] = "yes",
    ["yuppp"] = "yes",
    ["yepp"] = "yes",
    ["yepppp"] = "yes",
    
    -- Common Word Combinations
    ["imo"] = "in my opinion",
    ["imho"] = "in my honest opinion",
    ["tbh"] = "to be honest",
    ["tbf"] = "to be fair",
    ["ngl"] = "not gonna lie",
    ["fyi"] = "for your information",
    ["aka"] = "also known as",
    ["eg"] = "for example",
    ["ie"] = "that is",
    ["etc"] = "etcetera",
    ["w/"] = "with",
    ["w/o"] = "without",
    ["b/c"] = "because",
    
    -- More Number/Letter Substitutions
    ["l8r"] = "later",
    ["pot8o"] = "potato",
    ["tom8o"] = "tomato",
    ["sk8r"] = "skater",
    ["h8r"] = "hater",
    ["w8r"] = "waiter",
    ["4get"] = "forget",
    ["4got"] = "forgot",
    ["4given"] = "forgiven",
    ["2day"] = "today",
    ["2morrow"] = "tomorrow",
    ["2nite"] = "tonight",
    ["4ward"] = "forward",
    ["b4"] = "before",
    
    -- Extra Gaming Terms
    ["afk"] = "away from keyboard",
    ["brb"] = "be right back",
    ["gtg"] = "got to go",
    ["g2g"] = "got to go",
    ["ttyl"] = "talk to you later",
    ["gg"] = "good game",
    ["ggwp"] = "good game well played",
    ["glhf"] = "good luck have fun",
    ["ez"] = "easy",
    ["ezpz"] = "easy peasy",
    ["ezzz"] = "easy",
    ["1v1"] = "one versus one",
    ["clutch"] = "amazing play",
    
    -- More Slang/Informal
    ["sus"] = "suspicious",
    ["sussy"] = "suspicious",
    ["cap"] = "lie",
    ["no cap"] = "no lie",
    ["based"] = "agreeable",
    ["cringe"] = "embarrassing",
    ["yeet"] = "throw",
    ["yeeted"] = "threw",
    ["vibing"] = "relaxing",
    ["vibe"] = "atmosphere",
    ["hits different"] = "feels special",
    ["bet"] = "alright",
    ["finna"] = "going to",
    ["bouta"] = "about to",
    ["lowkey"] = "secretly",
    ["highkey"] = "obviously",
    ["deadass"] = "seriously",
    ["fr"] = "for real",
    ["frfr"] = "for real for real",
    ["ong"] = "on god",
    ["no shot"] = "no way",
    ["sheesh"] = "wow",
    
    -- More Emotional Expressions
    ["oof"] = "oh no",
    ["ooof"] = "oh no",
    ["oooof"] = "oh no",
    ["rip"] = "rest in peace",
    ["rippp"] = "rest in peace",
    ["f"] = "pay respects",
    ["sadge"] = "sad",
    ["pepehands"] = "very sad",
    ["kekw"] = "laughing",
    ["omegalul"] = "laughing hard",
    ["pog"] = "amazing",
    ["poggers"] = "amazing",
    ["monkas"] = "nervous",
    ["pepelaugh"] = "funny",
    
    -- Extra Corrections
    ["gonna"] = "going to",
    ["gotta"] = "got to",
    ["wanna"] = "want to",
    ["dunno"] = "don't know",
    ["gimme"] = "give me",
    ["lemme"] = "let me",
    ["tryna"] = "trying to",
    ["kinda"] = "kind of",
    ["sorta"] = "sort of",
    ["whatcha"] = "what are you",
    ["watchu"] = "what are you",
    ["ya"] = "you",
    ["chu"] = "you",
    ["dis"] = "this",
    ["dat"] = "that",
    ["dem"] = "them",
    ["dose"] = "those",
    ["dese"] = "these",
    -- Common shortcuts
    ["u"] = "you",
    ["ur"] = "your",
    ["urs"] = "yours",
    ["r"] = "are",
    ["n"] = "and",
    ["y"] = "why",
    ["k"] = "okay",
    ["bc"] = "because",
    ["cuz"] = "because",
    ["rn"] = "right now",
    ["tmr"] = "tomorrow",
    ["tn"] = "tonight",
    ["tdy"] = "today",
    ["alr"] = "alright",
    ["ppl"] = "people",
    
    -- Apologies and responses
    ["mb"] = "my bad",
    ["srry"] = "sorry",
    ["sry"] = "sorry",
    ["np"] = "no problem",
    ["yw"] = "you're welcome",
    ["ofc"] = "of course",
    
    -- Reactions
    ["lol"] = "laughing out loud",
    ["lmao"] = "laughing my ass off",
    ["rofl"] = "rolling on floor laughing",
    ["jk"] = "just kidding",
    
    -- Greetings
    ["hi"] = "hello",
    ["hey"] = "hello",
    ["sup"] = "what's up",
    ["wsp"] = "what's poppin'",
    ["wsg"] = "what's good",
    
    -- Common expressions
    ["idc"] = "I don't care",
    ["idm"] = "I don't mind",
    ["imo"] = "in my opinion",
    ["imho"] = "in my honest opinion",
    ["fr"] = "for real",
    ["ngl"] = "not gonna lie",
    ["fyi"] = "for your information",
    
    -- Contractions
    ["dont"] = "don't",
    ["doesnt"] = "doesn't",
    ["cant"] = "can't",
    ["wont"] = "won't",
    ["shouldnt"] = "shouldn't",
    ["couldnt"] = "couldn't",
    ["wouldnt"] = "wouldn't",
    ["wasnt"] = "wasn't",
    ["werent"] = "weren't",
    ["havent"] = "haven't",
    ["hasnt"] = "hasn't",
    ["didnt"] = "didn't",
    ["isnt"] = "isn't",
    ["arent"] = "aren't",
    
    -- Yes/No variations
    ["ye"] = "yes",
    ["yea"] = "yeah",
    ["nah"] = "no",
    ["nope"] = "no",
    
    -- Common misspellings
    ["gud"] = "good",
    ["git"] = "get",
    ["wat"] = "what",
    ["wut"] = "what",
    ["r u"] = "are you",
    ["pls"] = "please",
    ["plz"] = "please",
    ["thx"] = "thanks",
    ["thnx"] = "thanks",
    ["pic"] = "picture",
    ["pics"] = "pictures",
    ["sec"] = "second",
    ["mins"] = "minutes",
    ["msg"] = "message",
    
    -- Grammar corrections
    ["its"] = "it's",
    ["youre"] = "you're",
    ["theyre"] = "they're",
    ["whos"] = "who's",
    ["wheres"] = "where's",
    ["whats"] = "what's",
    ["whenre"] = "when're",
    ["howre"] = "how're",
    
    -- Common game terms
    ["ez"] = "easy",
    ["gg"] = "good game",
    ["wp"] = "well played",
    ["op"] = "overpowered",
        ["gud"] = "good",
    ["gudd"] = "good",
    ["gud"] = "good",
    ["gewd"] = "good",
    ["goot"] = "good",
    ["g00d"] = "good",
    ["kewl"] = "cool",
    ["kool"] = "cool",
    ["c00l"] = "cool",
    ["awsum"] = "awesome",
    ["awsome"] = "awesome",
    ["awesum"] = "awesome",
    ["awzum"] = "awesome",
    ["gr8"] = "great",
    ["gr8t"] = "great",
    ["grate"] = "great",
    ["noice"] = "nice",
    ["naice"] = "nice",
    ["nyc"] = "nice",
    ["nic"] = "nice",
    ["nyce"] = "nice",
    ["betta"] = "better",
    ["bettr"] = "better",
    ["betr"] = "better",
    ["wut"] = "what",
    ["wat"] = "what",
    ["wot"] = "what",
    ["whut"] = "what",
    ["wht"] = "what",
    ["y"] = "why",
    ["yy"] = "why",
    ["cuz"] = "because",
    ["bcuz"] = "because",
    ["bcz"] = "because",
    ["bcos"] = "because",
    ["bc"] = "because",
    ["b/c"] = "because",
    ["cos"] = "because",
    ["coz"] = "because",
    ["dat"] = "that",
    ["dis"] = "this",
    ["dese"] = "these",
    ["dose"] = "those",
    ["den"] = "then",
    ["den"] = "than",
    ["dey"] = "they",
    ["der"] = "there",
    ["tho"] = "though",
    ["doe"] = "though",
    ["thru"] = "through",
    ["thx"] = "thanks",
    ["thanx"] = "thanks",
    ["thnx"] = "thanks",
    ["tankz"] = "thanks",
    ["tanks"] = "thanks",
    ["ppl"] = "people",
    ["peeps"] = "people",
    ["ppls"] = "people",
    ["pplz"] = "people",
    ["sum1"] = "someone",
    ["sum"] = "some",
    ["sumthing"] = "something",
    ["sumtin"] = "something",
    ["sumn"] = "something",
    ["nuthin"] = "nothing",
    ["nuttin"] = "nothing",
    ["nuffin"] = "nothing",
    ["nutin"] = "nothing",
    ["sumwhere"] = "somewhere",
    ["sumware"] = "somewhere",
    ["everythin"] = "everything",
    ["evrything"] = "everything",
    ["evry"] = "every",
    ["everytin"] = "everything",
    ["errywhere"] = "everywhere",
    ["evrywhere"] = "everywhere",
    ["erywhere"] = "everywhere",
    ["prolly"] = "probably",
    ["prob"] = "probably",
    ["probly"] = "probably",
    ["probs"] = "probably",
    ["mayb"] = "maybe",
    ["mby"] = "maybe",
    ["mebbe"] = "maybe",
    ["definately"] = "definitely",
    ["deffo"] = "definitely",
    ["defo"] = "definitely",
    ["def"] = "definitely",
    ["dunno"] = "don't know",
    ["duno"] = "don't know",
    ["iuno"] = "I don't know",
    ["gonna"] = "going to",
    ["gunna"] = "going to",
    ["gon"] = "going to",
    ["wanna"] = "want to",
    ["wan"] = "want",
    ["hafta"] = "have to",
    ["gotta"] = "got to",
    ["gotsta"] = "got to",
    ["lemme"] = "let me",
    ["gimme"] = "give me",
    ["tryna"] = "trying to",
    ["finna"] = "fixing to",
    ["bouta"] = "about to",
    ["bout"] = "about",
    ["ima"] = "I'm going to",
    ["imma"] = "I'm going to",
    ["aint"] = "ain't",
    ["wasup"] = "what's up",
    ["wassup"] = "what's up",
    ["wussup"] = "what's up",
    ["sup"] = "what's up",
    ["aight"] = "alright",
    ["ight"] = "alright",
    ["aiight"] = "alright",
    ["rite"] = "right",
    ["ryt"] = "right",
    ["wright"] = "right",
    ["b4"] = "before",
    ["b4n"] = "bye for now",
    ["l8r"] = "later",
    ["l8"] = "late",
    ["str8"] = "straight",
    ["w8"] = "wait",
    ["m8"] = "mate",
    ["h8"] = "hate",
    ["sk8"] = "skate",
    ["g8"] = "gate",
    ["l8er"] = "later",
    ["2day"] = "today",
    ["2morrow"] = "tomorrow",
    ["2nite"] = "tonight",
    ["4got"] = "forgot",
    ["4get"] = "forget",
    ["4ward"] = "forward",
    ["4eva"] = "forever",
    ["sry"] = "sorry",
    ["srry"] = "sorry",
    ["soz"] = "sorry",
    ["sorrie"] = "sorry",
    ["pls"] = "please",
    ["plz"] = "please",
    ["plox"] = "please",
    ["pwease"] = "please",
    ["plez"] = "please",
    ["plzz"] = "please",
    ["plzzz"] = "please",
    ["thks"] = "thanks",
    ["tks"] = "thanks",
    ["ty"] = "thank you",
    ["thk"] = "thank",
    ["thnku"] = "thank you",
    ["thanku"] = "thank you",
    ["thankz"] = "thanks",
    ["thnks"] = "thanks",

    -- Gaming/Internet Slang
    ["ez"] = "easy",
    ["ezy"] = "easy",
    ["ezpz"] = "easy peasy",
    ["n00b"] = "noob",
    ["nub"] = "noob",
    ["newb"] = "noob",
    ["pwn"] = "own",
    ["pwned"] = "owned",
    ["rekt"] = "wrecked",
    ["rekd"] = "wrecked",
    ["rek"] = "wreck",
    ["git"] = "get",
    ["gits"] = "gets",
    ["gitin"] = "getting",
    ["gittin"] = "getting",
    ["hax"] = "hacks",
    ["haxor"] = "hacker",
    ["hackz"] = "hacks",
    ["h4x"] = "hacks",
    ["pro"] = "professional",
    ["proz"] = "professionals",
    ["skrub"] = "scrub",
    ["skrublord"] = "scrublord",
    ["leet"] = "elite",
    ["l33t"] = "elite",
    ["1337"] = "elite",
    
    -- Numbers as Letters
    ["m3"] = "me",
    ["u2"] = "you too",
    ["r3kt"] = "wrecked",
    ["n1"] = "nice one",
    ["2u2"] = "to you too",
    ["4u"] = "for you",
    ["m8t"] = "mate",
    
    -- Emotional/Reaction
    ["lel"] = "lol",
    ["kek"] = "lol",
    ["lawl"] = "lol",
    ["lulz"] = "lols",
    ["rofl"] = "rolling on floor laughing",
    ["rofls"] = "rolling on floor laughing",
    ["roflmao"] = "rolling on floor laughing my ass off",
    ["lmfao"] = "laughing my fucking ass off",
    ["haha"] = "ha ha",
    ["hehe"] = "he he",
    ["xd"] = "XD",
    ["dx"] = "DX",
    ["omfg"] = "oh my fucking god",
    ["wtf"] = "what the fuck",
    ["tf"] = "the fuck",
    ["ffs"] = "for fuck's sake",
    ["smh"] = "shaking my head",
    ["nvm"] = "never mind",
    -- Miscellaneous
    ["oh"] = "oh,",
    ["gramer"] = "grammar",
    ["grammer"] = "grammar",
    ["gremmer"] = "grammar",
    ["anymor"] = "anymore",
    ["amerika"] = "america"
}

-- Rest of the code remains the same...


local Questions = {
    "mean",
    "ask",
    "you",
    "care",
    "is",
    "what",
    "it",
    "question",
    "here",
    "where"
}

-- Utility functions
local function GrabFirstLetter(Text)
    return string.sub(Text,1,1)
end

local function GrabLowercased(Text)
   return string.sub(Text,2)
end

local function AddStrings(String1,String2)
    return tostring(String1..String2)
end

local function GrabLastCharacter(Text)
    return string.sub(Text,string.len(Text),string.len(Text))
end

local function ConvertWords(Table)
    for i, v in pairs(Table) do
        v = v:lower()
        if Corrections[v] then
            pcall(function()
                if Table[i] == Table[1] then
                    local Corrected = Corrections[v]
                    local FirstLetter = GrabFirstLetter(Corrected)
                    local Lowercased = GrabLowercased(Corrected)
                    Table[i] = FirstLetter:upper()..Lowercased
                else
                    Table[i] = Corrections[v]
                end
            end)
        end
    end
end

-- Set up chat input based on chat version
local ChatInputTextBox
if game.TextChatService.ChatVersion == Enum.ChatVersion.TextChatService then
    ChatInputTextBox = game:GetService("CoreGui").ExperienceChat.appLayout.chatInputBar.Background.Container.TextContainer.TextBoxContainer.TextBox
else
    ChatInputTextBox = game:GetService("Players").LocalPlayer.PlayerGui.Chat.Frame.ChatBarParentFrame.Frame.BoxFrame.Frame.ChatBar
end

-- Chat function
local function chat(txt)
    local Text = txt
    
    -- First letter converting
    local FirstLetter = GrabFirstLetter(Text)
    local Lowercased = GrabLowercased(Text)
    local CapitalText = AddStrings(FirstLetter:upper(),Lowercased)
    
    -- Word correction
    local Table = string.split(CapitalText," ")
    ConvertWords(Table)  
    local Final = ""
    
    -- Adding all words back
    for i, v in pairs(Table) do
        if Table[i] == Table[1] then
            Final = Final..v
        else
            Final = Final.." "..v
        end
    end
    
    -- Adding punctuation
    local LastCharacter = GrabLastCharacter(Final)
    if LastCharacter == "!" or LastCharacter == "." or LastCharacter == "?" then
        txt = Final
    else
        if table.find(Questions,Table[#Table]:lower()) then
            txt = Final.."?"
        else
            txt = Final.."."
        end
    end
    
    if game:GetService("ReplicatedStorage"):FindFirstChild("DefaultChatSystemChatEvents") then
        game:GetService("ReplicatedStorage").DefaultChatSystemChatEvents.SayMessageRequest:FireServer(txt,"All")
    else
        game:GetService("TextChatService").TextChannels.RBXGeneral:SendAsync(txt)
    end
end

-- Handle chat input
ChatInputTextBox.FocusLost:Connect(function(enterPressed)
    if enterPressed then
        chat(ChatInputTextBox.Text)
        ChatInputTextBox.Text = ""
    end
end)
